### PR TITLE
Refactor SearchInput to functional component with hooks

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
**What kind of change does this PR introduce?**
Update components to functional components with Hooks

**What is the current behavior?**
Class components

**What is the new behavior?**
Functional components with Hooks

**Checklist**:

- [ ] Documentation N/A
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: packages/homepage/src/screens/home/NPMFeature/SearchInput/Hit.js [hunk 1]
```diff
@@ -38,37 +37,31 @@ const Count = styled.div`
...
-      this.el,
-      0.1,
-      { opacity: 0 },
-      { opacity: 1 }
-    );
-  }
+const Hit = ({ hit: { count, value } }) => {
+  const el = useRef(null);
+  const tl = useRef(
+    new TimelineMax().fromTo(el.current, 0.1, { opacity: 0 }, { opacity: 1 })
```
Comment (SaraVieira) [inline packages/homepage/src/screens/home/NPMFeature/SearchInput/Hit.js]:
```suggestion
  let tl;

  useEffect(() => {
    tl = new TimelineMax().fromTo(
      el.current,
      0.1,
      { opacity: 0 },
      { opacity: 1 }
    );
  }, []);
```

**Comment by SaraVieira** [comment]
Hey!

I get an error on running it because sometimes the current target does not exist and I added a comment that fixes it by only checking the current target on component mount :)

**Comment by MichaelDeBoey** [comment]
@SaraVieira Needed to create a `ref` for `tl`, otherwise I got the following warning from ESLint:
```sh
44:10  warning  Assignments to the 'tl' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect  react-hooks/exhaustive-deps
```

Also added `eslint-plugin-react-hooks` dependency and `react-hooks/rules-of-hooks` and `react-hooks/exhaustive-deps` rules, so we can take advantage of them.

**Comment by SaraVieira** [comment]
Interesting error but it looks good, works well in your machine?

**Comment by MichaelDeBoey** [comment]
@SaraVieira Works just like before on my machine

**Comment by SaraVieira** [comment]
Awesome!! Thank you again!

**Comment by MichaelDeBoey** [comment]
With pleasure 🙂

**Pull request merged**